tests/integration-tests/tests/api/test_api_client.py

```diff
@@ -136,3 +142,62 @@ def _test_delete_cluster(client, cluster, region):
     response = client.list_clusters(region=region)
     cluster_names = [c["cluster_name"] for c in response["items"]]
     assert_that(cluster_names).does_not_contain(cluster_name)
+
+
+@pytest.mark.usefixtures("instance")
+def test_custom_image(request, region, os, pcluster_config_reader, api_client):
+    base_ami = retrieve_latest_ami(region, os)
+
+    config_file = pcluster_config_reader(config_file="image.config.yaml", parent_image=base_ami)
+    with open(config_file) as config_file:
+        config = config_file.read()
+
+    stack_name = generate_stack_name("integ-tests", request.config.getoption("stackname_suffix"))
+    client = image_operations_api.ImageOperationsApi(api_client)
+
+    with soft_assertions():
+        resp = _test_build_image(client, config, region, stack_name)
+        LOGGER.info("Build image response: %s", resp)
+        image = resp["image"]
+        image_id = image["image_id"]
+        assert_that(image_id).is_equal_to(stack_name)
+
+        # CFN stack is deleted as soon as image is available
+        _cloudformation_wait(region, stack_name, "stack_delete_complete")
+
+        _test_describe_image(client, image, region)
+        _test_list_images(client, image, region)
+        _delete_image(client, image, region)
+
+
+def _test_build_image(client, image_config, region, image_id):
+    image_config_data = base64.b64encode(image_config.encode("utf-8")).decode("utf-8")
+    body = BuildImageRequestContent(image_config_data, image_id, region=region)
+    return client.build_image(body)
+
+
+def _test_describe_image(client, image, region):
+    image_id = image["image_id"]
+    response = client.describe_image(image_id, region=region)
+    LOGGER.info("Describe image response: %s", response)
+    assert_that(response.image_id).is_equal_to(image_id)
+
+
+def _test_list_images(client, image, region):
+    image_id = image["image_id"]
+
+    # Despite being a paginated call, available images are returned all in the first page
+    response = client.list_images(image_status="AVAILABLE", region=region)
+    LOGGER.info("List images response: %s", response)
+    image_ids = [item["image_id"] for item in response["items"]]
```

Comment:
let's use sets rather than list when we need to check that an item is present. same elsewhere

Reply:
I will re-write this call to unwrap the result, since if we want to check the list of pending images right after a build_image, we cannot rely on getting the target image in the first page of results.